src/jira.ts

```diff
@@ -0,0 +1,2303 @@
+						.setTitle(`${req.body.issue.key}: ${req.body.issue.fields.summary}`)
+						.setColor('#0052cc')
+						.setDescription(req.body.issue.fields.description ?? 'No description available')
+						.addField('Status', req.body.issue.fields.status.name)
+						.addField('Source', `[link](${req.body.issue.fields[config.jira.fields.videoLink]})`)
+						.setURL(`${config.jira.url}/projects/${req.body.issue.fields.project.key}/issues/${req.body.issue.key}`);
+
+					const newMsg = await newChannel.send({
+						content: 'New project ready for release',
+						embeds: [embed],
+					}).catch((err) => {
+						const eventId = Sentry.captureException(err);
+						logger.error(`Encountered error while sending message (${eventId})`);
+						logger.error(err);
+						encounteredError = true;
+					});
+					if (encounteredError) return;
+
+					link.discordMessageId = newMsg!.id;
+					link.status = req.body.issue.fields.status.name;
+					link.lastUpdate = new Date();
+					link.hasAssignment = 0;
+
+					link.save((err) => {
+						if (err) {
+							const eventId = Sentry.captureException(err);
+							logger.error(`Encountered error saving issue link (${eventId})`);
+							logger.error(err);
+						}
+					});
+				} else {
+					let user: any | undefined;
+					if (req.body.issue.fields.assignee !== null) {
+						const oauthUserRes = await axios.get(`${config.oauthServer.url}/api/userByJiraKey`, {
+							params: { key: req.body.issue.fields.assignee.key },
+							auth: {
+								username: config.oauthServer.clientId,
+								password: config.oauthServer.clientSecret,
+							},
+						}).catch((err) => {
+							const eventId = Sentry.captureException(err);
+							logger.error(`Encountered error while fetching user from OAuth (${eventId})`);
+							encounteredError = true;
+						});
+						if (encounteredError) return;
+
+						user = oauthUserRes!.data;
+						link.hasAssignment = (1 << 0);
+					} else {
+						link.hasAssignment = 0;
+					}
+
+					const embed = new MessageEmbed()
+						.setTitle(`${req.body.issue.key}: ${req.body.issue.fields.summary}`)
+						.setColor('#0052cc')
+						.setDescription(req.body.issue.fields.description ?? 'No description available')
+						.addField('Status', req.body.issue.fields.status.name!)
+						.addField('Assignee', user ? `<@${user._id}>` : 'Unassigned')
+						.addField('Source', `[link](${req.body.issue.fields[config.jira.fields.videoLink]})`)
+						.setFooter({ text: `Due date: ${req.body.issue.fields.duedate || 'unknown'}` })
+						.setURL(`${config.jira.url}/projects/${req.body.issue.fields.project.key}/issues/${req.body.issue.key}`);
+
+					const newMsg = await newChannel.send({
+						embeds: [embed],
+						components: [row],
+					}).catch((err) => {
+						const eventId = Sentry.captureException(err);
+						logger.error(`Encountered error while sending message (${eventId})`);
+						logger.error(err);
+					});
+
+					link.discordMessageId = newMsg?.id ?? undefined;
```

Comment:
```suggestion
					link.discordMessageId = newMsg?.id;
```
Should be undefined already? Should change this unless TS gives an error about this.